clash-ghc/src-ghc/Clash/GHC/Evaluator/Primitive.hs

```diff
@@ -4370,13 +4373,24 @@ ghcPrimStep tcm isSubj pInfo tys args mach = case primName pInfo of
          0 -> reduce (valToTerm z)
          _ -> let (tyArgs,_)  = splitFunForallTy ty
                   (tyArgs',_) = splitFunForallTy (Either.rights tyArgs !! 2)
-                  TyConApp snatTcNm _ = tyView (Either.rights tyArgs' !! 0)
+                  ubpT = Either.rights tyArgs' !! 0
+                  fTVs = Lens.toListOf typeFreeVars ubpT
+                  Just tvN = List.find ((== "n") . nameOcc . varName) fTVs
+                  subst0 = extendTvSubst (mkSubst is0) tvN k'ty
+                  Just tvK = List.find ((== "k") . nameOcc . varName) fTVs
+                  subst1 = extendTvSubst subst0 tvK (LitTy (NumTy k'))
```

Comment:
`tvN` and `tvK` are probably in `Either.lefts tyArgs'`, no need to "find" them.

Reply:
`Either.lefts tyArgs'` only contains `tvN`, while `fTVs`holds both `tvN` and `tvK`, but nothing more. I updated it in the sense that I take `tvN` from `Either.lefts tyArgs'` now and then pick `tvK` as the first non-`tvN` element in `fTVs`.